package-readmes/storesShareasale.md

```diff
@@ -0,0 +1,156 @@
+```
+
+The file contains the following function: 
+
+
+
+*   **reportSaleToShareASale()**
+
+    Creates a sales report and sends it to ShareASale. You can call this function when a customer payment is validated. 
+
+
+    Syntax:
+
+
+    ```js
+    export function reportSaleToShareASale(sscid: string, orderId: number, buyerId: number, reportOptions: object):Promise<string>
+    ```
+
+
+    Parameters:
+
+*   **`sscid`:** Affiliate ID obtained from the retrieveStoredAffiliateId() function in the share-a-sale-public.js file.
+*   **`orderId`:** **Wix Store** order ID. Note that each order ID can only be used for a single report. Additional requests with the same order ID will be ignored.
+*   **`buyerId`:** Contact ID of the buyer. Can be obtained by calling the Velo `[appendOrCreateContact()` API](https://www.wix.com/velo/reference/wix-crm/contacts-obj/appendorcreatecontact). 
+*   **`reportOptions`:** An object containing optional parameters to report to ShareASale. The object can only contain the following keys. The API call will be rejected if other keys are provided.
+        *  **`currency`:** The 3-letter [ISO-4217](https://www.iban.com/currency-codes) currency code used by the customer. For example, EUR for Euros. Defaults to USD if the value is blank, or if the specified currency code is invalid.
+        *  **`commissionOverridePercentage`:** Number of the commission percentage you want in the report.
+        *  **`commissionOverrideAmount`:** Number of the commission amount you want in the report.
+
+    Returns:
+
+    A promise that resolves to a string with either an error message or a success message from ShareASale. 
+
+
+#### Public File 
+
+This package contains the following public file:
+
+
+##### share-a-sale-public.js 
+The code in this file contains functions needed to detect when a site visitor is using an affiliate ID, and to store and to retrieve an affiliate ID. 
+
+To use the functions below in your code, import it with the following syntax:
+
+```js
+import { <functionName> } from '@velo/wix-stores-shareasale'
+```
+
+The file contains the following functions: 
+
+
+
+*   **detectAndStoreAffiliateId()**
+
+    Checks if a site visitor is accessing your website through an affiliate, and stores that affiliate's ID in **[Wix Session Storage](https://www.wix.com/velo/reference/wix-storage/session)**. This function should be called in the **[masterPage.js](https://support.wix.com/en/article/velo-where-do-i-put-my-code#i-want-some-code-to-run-on-every-page)** code file to cover all pages of your site.
+
+
+    Syntax:
+
+
+    ```js
+    export function detectAndStoreAffiliateId()
+    ```
+
+    **Note:** The site's [$w.onReady()](https://www.wix.com/velo/reference/$w/onready) function runs twice. To prevent the function from being called twice, add the following if statement before calling the function in the **masterPage.js** file. [Learn more.](https://www.wix.com/velo/reference/wix-window/rendering-obj/env)
```

Comment:
code voice here

```suggestion
    **Note:** The site's [$w.onReady()](https://www.wix.com/velo/reference/$w/onready) function runs twice. To prevent the function from being called twice, add the following `if` statement before calling the function in the **masterPage.js** file. [Learn more.](https://www.wix.com/velo/reference/wix-window/rendering-obj/env)
```